api/jobs.go

```diff
@@ -433,6 +435,9 @@ func (u *UpdateStrategy) Copy() *UpdateStrategy {
 		copy.Canary = intToPtr(*u.Canary)
 	}
 
+	// boolPtrs need to preserve nil unset value
+	copy.AutoPromote = u.AutoPromote
```

Comment:
I don't quite follow this comment and why the copying here is different?

I presume that the rest of assignments used `boolToPtr` to avoid having two structures share the same reference when deep copying the structure; but I don't know for sure and we don't really do reference  manipulation here afaik.

Reply:
good catch, I thought about that wrong. copy and merge should match the old behavior